source/UIABrowseMode.py

```diff
@@ -185,9 +187,15 @@ def UIAControlQuicknavIterator(itemType,document,position,UIACondition,direction
 	if direction=="up":
 		walker=UIAHandler.handler.clientObject.createTreeWalker(UIACondition)
 		element=position.UIAElementAtStart
-		element=walker.normalizeElement(element)
-		if element and not UIAHandler.handler.clientObject.compareElements(element,document.rootNVDAObject.UIAElement) and not UIAHandler.handler.clientObject.compareElements(element,UIAHandler.handler.rootElement):
-			yield itemClass(itemType,document,element)
+		while element:
+			element=walker.normalizeElement(element)
+			if not element or UIAHandler.handler.clientObject.compareElements(element,document.rootNVDAObject.UIAElement) or UIAHandler.handler.clientObject.compareElements(element,UIAHandler.handler.rootElement):

```

Comment:
It would be nice if this line was split onto multiple lines, or each part given a representative name to explain it.